justfile

```diff
@@ -86,10 +86,12 @@ sol-test:
     forge test
 
 # Deploy contracts to local blockchain for development and testing
-dev-deploy url="http://localhost:8545" mnemonics="test test test test test test test test test test test junk" num_blocks_per_epoch="10" num_init_validators="5":
-    MNEMONICS="{{mnemonics}}" forge script contracts/test/LightClientTest.s.sol:DeployLightClientTestScript \
-    --sig "run(uint32 numBlocksPerEpoch, uint32 numInitValidators)" {{num_blocks_per_epoch}} {{num_init_validators}} \
-    --fork-url {{url}} --broadcast
+dev-deploy url="" mnemonics="" num_blocks_per_epoch="10" num_init_validators="5" admin="0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266" private_key="":
+    MNEMONICS="{{mnemonics}}" forge script contracts/test/mocks/DeployLightClientTestScript.sol:DeployLightClientTestScript \
+    --sig "run(uint32 numBlocksPerEpoch, uint64 numInitValidators, address owner)" {{num_blocks_per_epoch}} {{num_init_validators}} {{admin}} \
+    --fork-url {{url}} --broadcast \
+    --private-key {{private_key}}
```

Comment:
Does this work if no private key is specified?

To make optional arguments we can use `*args` like on line 11 and pass them to `forge script`.

```
down *args:
    docker compose down {{args}}
 ```

Reply:
It didn't without the private key for me, I got this error:
`No associated wallet for addresses: {0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266}. Unlocked wallets: []`

and used this solution https://github.com/Cyfrin/foundry-full-course-f23/discussions/866#discussioncomment-7000638

Reply:
Sorry I think I wasn't explicit enough.

I think currently the way this just command is written if one runs it without specifying a private key the command line will turn into `forge script ... --private-key ""` which I think is not a valid `forge script` command because  I would expect that `--private-key` expects a private key to follow and the empty string isn't a valid private key.

That's why I suggested putting `*args` there instead so that `dev-deploy` can accept additional arguments and hand them to `forge script` via `{{args}}`.

Reply:
Ok thank you for clarifying. I spoke to @alxiong about this too and we agreed to create a separate ticket to change the deployment implementation so we'll leave this as is for this ticket.

Reply:
yes, there's already a ticket: https://github.com/EspressoSystems/espresso-sequencer/issues/1177

i'll notify you guys when I have a PR.